From 5d9a07a1e805bc019a2a32e611ca2503197e3704 Mon Sep 17 00:00:00 2001
From: Mark Rushakoff <mark@strange.love>
Date: Fri, 16 Aug 2024 15:39:48 -0400
Subject: [PATCH 7/7] fix: default client context's output to cmd.OutOrStdout

If unset, it uses os.Stdout, and we cannot capture that output from
tests involving commands. While this probably doesn't fix every use, it
fixes the "tx staking create-validator" output, and probably prevents
some other future commands we will test against.
---
 client/cmd.go | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/client/cmd.go b/client/cmd.go
index 7ea1f49e09..a4a1b94e2e 100644
--- a/client/cmd.go
+++ b/client/cmd.go
@@ -345,7 +345,8 @@ func GetClientQueryContext(cmd *cobra.Command) (Context, error) {
 // - client.Context field pre-populated & flag not set: uses pre-populated value
 // - client.Context field pre-populated & flag set: uses set flag value
 func GetClientTxContext(cmd *cobra.Command) (Context, error) {
-	ctx := GetClientContextFromCmd(cmd)
+	ctx := GetClientContextFromCmd(cmd).WithOutput(cmd.OutOrStdout())
+
 	return readTxCommandFlags(ctx, cmd.Flags())
 }
 
-- 
2.44.0

